redhat-cop/casl-ansible
# Add an optional custom pre-install playbook

#### What does this PR do?

This adds a way to specify an optional playbook that will be run at the end of the
pre-install section (i.e. after the openstack resources are up but
before the openshift installation).

It's useful for uploading the LDAP CA to the master nodes and any other custom pre-install configuration.

I'm sot sure this is necessarily the best way to go, but it's very simple and flexible. I haven't updated the docs or anything in case this gets closed/done differently yet.

#### How should this be manually tested?

Write a new playbook that does something on the nodes (say copies a file over). Run the end-to-end deployment with `-e openstack_custom_pre_install_playbook=<path to the playbook>`. Once it completes, verify that the custom playbook was executed.

#### Is there a relevant Issue open for this?

No

#### Who would you like to review this?
cc: @redhat-cop/casl
